pytun_pmd3/wintun.py

```diff
@@ -313,6 +326,26 @@ def read(self) -> bytes:
 
         return packet_data
 
+    async def async_read(self) -> bytes:
+        while True:
+            size = DWORD()
+            packet_ptr = wintun.WintunReceivePacket(self.session, byref(size))
+            if packet_ptr:
+                packet_data = string_at(packet_ptr, size.value)
+                wintun.WintunReleaseReceivePacket(self.session, packet_ptr)
+                if (packet_data[0] >> 4) == 6:
+                    return packet_data
+                return b''
+
+            last_error = get_last_error()
+            if last_error == ERROR_NO_MORE_ITEMS:
+                wait_event = wintun.WintunGetReadWaitEvent(self.session)
```

Comment:
You can create the wait event once in the `up` method after creating the session.

Reply:
done